src/DefaultBuilder/src/WebApplicationBuilder.cs

```diff
@@ -166,6 +173,12 @@ private void ConfigureApplication(WebHostBuilderContext context, IApplicationBui
             }
         }
 
+        if (Authentication is WebApplicationAuthenticationBuilder webAuthBuilder && webAuthBuilder.IsAuthenticationConfigured is true)
+        {
+            app.UseAuthentication();
+            app.UseAuthorization();
```

Comment:
Can we add something to `WebApplicationTests` (or another test class since that's already kinda big) to verify we register services and add middleware when methods are called on the `WebApplicationAuthenticationBuilder`? And another test verifying that just accessing the `Authentication` getter doesn't do this?

Reply:
Yep, will add this to verify changes for https://github.com/dotnet/aspnetcore/pull/41520#discussion_r883961906